File: 2021/python/aoc1.py
import numpy as np

input_path = '../inputs/day_1.txt'
lines = open(input_path).read().split('\n')
x = np.asarray(list(map(float,lines)))

#part 1
p1 = sum(np.diff(x) > 0)
print(p1)
#part 2
p2 = sum(np.diff(np.convolve(x, [1,1,1],'valid' )) > 0)
print(p2)

File: 2021/matlab/aoc1.m
run(fullfile(get_input_root(),'day_1.m'))
% part 1
p1 = sum(diff(x)>0)
% part 2
p2 = sum(diff(conv(x,[1,1,1],'valid'))>0)

assert(p1 == 1692)
assert(p2 == 1724)